core/message.go

```diff
@@ -194,7 +202,12 @@ func (msg *Message) CloneOriginal() *Message {
 func (msg *Message) FreezeOriginal() {
 	msg.orig.payload = getPayloadCopy(msg.data.payload)
 	msg.orig.streamID = msg.data.streamID
-	msg.orig.Metadata = msg.data.Metadata.Clone()
+
+	if msg.data.Metadata == nil {
```

Comment:
well, there is the theoretical case that data.Metadata changes from a map to nil but that should never happen. And the next PR (delayed freeze) will change Metadata behavior anyway.